tests/ui/traits/next-solver/diagnostics/point-at-failing-nested.stderr

```diff
@@ -0,0 +1,23 @@
+error[E0277]: the trait bound `(): Foo` is not satisfied
+  --> $DIR/point-at-failing-nested.rs:22:17
+   |
+LL |     needs_foo::<()>();
+   |                 ^^ the trait `Bar` is not implemented for `()`, which is required by `(): Foo`
```

Comment:
This used to say:
> the trait `Sized` is not implemented for `()`
which is obviously incorrect, but due to `?0: Sized` being a nested goal.

This wouldve been fixed by #124690, but then we wouldn't be reporting a nested goal as the failure cause at all.